src/components/common/connectionDialog/internetConnection.js

```diff
@@ -31,6 +21,19 @@ const InternetConnection = props => {
   const isConnectionWeb3 = useConnectionWeb3()
   const isConnectionGun = useConnectionGun()
   const [showContent, setShowContent] = useState(false)
+  const showDialogWindow = useCallback(
```

Comment:
you need to define the [parameters] the callback depends on, you want the callback to be re-defined only when they change
https://reactjs.org/docs/hooks-reference.html#usecallback

Reply:
@sirpy We should use the same parameter as for useEffect?

Reply:
@sirpy 
I discussed this with @AlexeyKosinski and we think that empty array should be passed as a second parameter to useCallback what will remember the showDialogWindow fn only once as we don't need to redefine it anymore as no any other param does not effect on it.

Reply:
@sirpy I pushed new changes

Reply:
1. there's no reason to put message as a state variable, it should just be an input to showDialogWindow
2. setShowContent should be called inside showDialogWindow as the purpose of debounce is to show the error dialog only after a while. maybe its better to change the name to "showDisconnect"
3. the variables/functions that showDialogWindow uses are showDialog and setShowDisconnect so the array of dependencies should be [showDialog,setShowDisconnect]

Reply:
ok